core/keyserver/keyserver.go

```diff
@@ -276,7 +277,93 @@ func (s *Server) ListEntryHistory(ctx context.Context, in *pb.ListEntryHistoryRe
 // ListUserRevisions returns a list of revisions covering a period of time.
 func (s *Server) ListUserRevisions(ctx context.Context, in *pb.ListUserRevisionsRequest) (
 	*pb.ListUserRevisionsResponse, error) {
-	return nil, status.Error(codes.Unimplemented, "not implemented")
+	offset := int64(0)
+	lastVerified := in.GetLastVerifiedTreeSize()
+	if in.GetPageToken() != "" {
+		token := &rtpb.ListUserRevisionsToken{}
+		if err := DecodeToken(in.GetPageToken(), token); err != nil {
+			glog.Errorf("invalid page token %v: %v", in.GetPageToken(), err)
+			return nil, status.Errorf(codes.InvalidArgument, "Invalid page_token provided")
+		}
+		// last_verified_tree_size and page_token are allowed to change between paginated requests.
+		// Clear them here both for comparison and for encoding next_page_token in the response.
+		in.LastVerifiedTreeSize = 0
+		in.PageToken = ""
+		if !proto.Equal(in, token.Request) {
+			return nil, status.Errorf(codes.InvalidArgument, "Request fields changed during pagination")
+		}
+		offset = token.RevisionsReturned
+	}
+
+	// Lookup log and map info.
+	directoryID := in.GetDirectoryId()
+	if directoryID == "" {
+		return nil, status.Errorf(codes.InvalidArgument, "Please specify a directory_id")
+	}
+	d, err := s.directories.Read(ctx, directoryID, false)
+	if err != nil {
+		glog.Errorf("adminstorage.Read(%v): %v", directoryID, err)
+		return nil, status.Errorf(codes.Internal, "Cannot fetch directory info")
+	}
+
+	// Fetch latest revision.
+	sth, consistencyProof, err := s.latestLogRootProof(ctx, d, lastVerified)
+	if err != nil {
+		return nil, err
+	}
+	currentRevision, err := mapRevisionFor(sth)
```

Comment:
"lastRevision" or "endRevision" seems more correct wdyt?

Reply:
We're fetching the newest revision, so I think "current" makes sense.

Reply:
Renamed.